sputnikdao-factory2/src/lib.rs

```diff
@@ -432,4 +433,69 @@ mod tests {
             vec![format!("test.{}", accounts(0)).parse().unwrap()]
         );
     }
+
+    //              #################################              //
+    //              #    Factory ownership tests    #              //
+    //              #################################              //
+
+    #[test]
+    fn test_factory_can_get_current_owner() {
+        let mut context = VMContextBuilder::new();
+        testing_env!(context
+            .current_account_id(alice())
+            .predecessor_account_id(alice())
+            .attached_deposit(10)
+            .build());
+        let factory = SputnikDAOFactory::new();
+
+        assert_eq!(factory.get_owner(), alice());
+    }
+
+    #[test]
+    #[should_panic]
+    fn test_factory_fails_setting_owner_from_not_owner_account() {
+        let mut context = VMContextBuilder::new();
+        testing_env!(context
+            .current_account_id(alice())
+            .predecessor_account_id(carol())
+            .attached_deposit(10)
+            .build());
+        let factory = SputnikDAOFactory::new();
+
+        factory.set_owner(bob());
+    }
+
+    #[test]
+    fn test_owner_can_be_a_dao_account() {
+        let mut context = VMContextBuilder::new();
+        testing_env!(context
+            .current_account_id(bob())
+            .predecessor_account_id(bob())
+            .attached_deposit(20)
+            .build());
+        let mut factory = SputnikDAOFactory::new();
+
+        factory.create(bob(), "{}".as_bytes().to_vec().into());
+
+        factory.set_owner(alice());
+
+        assert_eq!(factory.get_owner(), alice())
```

Comment:
This logic is wrong - it needs to use the DAO account in the `set_owner` method, then check that the DAO account is owner in `get_owner`.
Since you're using bob as the name, it will be something like `bob.sputnik-dao.near` as the DAO account. 
So make sure to change `set_owner` to be the actual full dao name

Reply:
yes, all fixed - thanks!